pkg/file/buffer_test.go

```diff
@@ -113,3 +116,134 @@ func testChunkPipe(t *testing.T) {
 		}
 	}
 }
+
+func TestCopyBuffer(t *testing.T) {
+	readBufferSizes := []int{
+		64,
+		1024,
+		swarm.ChunkSize,
+	}
+	dataSizes := []int{
+		1,
+		64,
+		1024,
+		swarm.ChunkSize - 1,
+		swarm.ChunkSize,
+		swarm.ChunkSize + 1,
+		swarm.ChunkSize * 2,
+		swarm.ChunkSize*2 + 3,
+		swarm.ChunkSize * 5,
+		swarm.ChunkSize*5 + 3,
+		swarm.ChunkSize * 17,
+		swarm.ChunkSize*17 + 3,
+	}
+
+	testCases := []struct {
+		readBufferSize int
+		dataSize       int
+	}{}
+
+	for i := 0; i < len(readBufferSizes); i++ {
+		for j := 0; j < len(dataSizes); j++ {
+			testCases = append(testCases, struct {
+				readBufferSize int
+				dataSize       int
+			}{readBufferSizes[i], dataSizes[j]})
+		}
+	}
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("buf_%-4d/data_size_%d", tc.readBufferSize, tc.dataSize), func(t *testing.T) {
+			// https://golang.org/doc/faq#closures_and_goroutines
+			readBufferSize := tc.readBufferSize
```

Comment:
is this really needed? these nested tests are indeed run in different goroutines but the for iterator should not progress to the next test before the previous one has not finished no?

Reply:
At first I though the same, however the race detector complained.

It might not have been needed for both variables, however I figured to have them both copied.